src/components/BarChart.js

```diff
@@ -1,299 +1,351 @@
-    const y = this.props.height - height + this.props.y;
-    const shift = this.props.value > 0 ? '3px' : '-3px';
-    const style = {
-      fill: this.state.hovering && this.props.primaryColor ? this.props.primaryColor : this.props.color,
-      cursor: 'pointer',
-      transform: 'translateY(' + shift + ')'
-    };
-
     return (
-      <rect
-        height={height}
-        onClick={this.props.onClick.bind(null, this.props.value, this.props.label)}
-        onMouseOut={this._handleMouseOut}
-        onMouseOver={this._handleMouseOver.bind(null, this.props.label, this.props.value, this.props.x, y)}
-        rx={3}
-        ry={3}
-        style={style}
-        width={this.props.width}
-        x={this.props.x}
-        y={y}
+      <path
+        d={this._drawPath({
+          x: this.props.x,
+          y: this.props.y,
+          width: this.props.width,
+          height: this.props.height,
+          value: this.props.value,
+          radius: this.props.radius
+        })}
+        onClick={this.props.onClick}
+        onMouseOut={this.props.onMouseOut}
+        onMouseOver={this.props.onMouseOver}
+        ref={ref => {
+          this.bar = ref;
+        }}
+        style={this.props.style}
       />
     );
   }
 });
 
-
 const BarChart = React.createClass({
   propTypes: {
     animateOnHover: React.PropTypes.bool,
+    barRadius: React.PropTypes.number,
     data: React.PropTypes.array.isRequired,
     height: React.PropTypes.number,
-    labelStyle: React.PropTypes.object,
+    margin: React.PropTypes.shape({
+      top: React.PropTypes.number,
+      right: React.PropTypes.number,
+      bottom: React.PropTypes.number,
+      left: React.PropTypes.number
+    }),
     onClick: React.PropTypes.func,
     onHover: React.PropTypes.func,
-    primaryColor: React.PropTypes.string,
-    tooltipFormat: React.PropTypes.string,
-    tooltipStyle: React.PropTypes.object,
-    width: React.PropTypes.number
+    showTooltips: React.PropTypes.bool,
+    style: React.PropTypes.object,
+    tooltipFormat: React.PropTypes.func,
+    width: React.PropTypes.number,
+    xAxis: React.PropTypes.element,
+    yAxis: React.PropTypes.element
   },
 
   getDefaultProps () {
     return {
       animateOnHover: false,
+      barRadius: 3,
       height: 300,
+      margin: {
+        top: 20,
+        right: 20,
+        bottom: 40,
+        left: 20
+      },
       onClick: () => {},
       onHover: () => {},
-      primaryColor: StyleConstants.Colors.PRIMARY,
-      tooltipFormat: '$0,0.00',
-      width: 500
+      style: {},
+      tooltipFormat: (val) => val,
+      width: 500,
+      showTooltips: true
     };
   },
 
   getInitialState () {
     return {
-      hovering: false
+      hoveringObj: {},
+      clickedData: {}
     };
   },
 
   shouldComponentUpdate (nextProps, nextState) {
     return !_isEqual(nextProps, this.props) || !_isEqual(nextState, this.state);
```

Comment:
Since that is more the public API, I don't think we can guarantee that those functions aren't changing? That's my thinking anyways...thoughts?